Client/game_sa/CModelInfoSA.cpp

```diff
@@ -1728,6 +1734,69 @@ void CModelInfoSA::MakeVehicleAutomobile(ushort usBaseID)
     CopyStreamingInfoFromModel(usBaseID);
 }
 
+bool GetFirstAtomic(RpAtomic* atomic, void* data)
+{
+    RpAtomic** firstAtomicPtr = (RpAtomic**)data;
+
+    if (*firstAtomicPtr == nullptr)
+        *firstAtomicPtr = atomic;
+
+    return false;
+}
+
+bool CModelInfoSA::MakeAtomicModel(ushort usModelId)
+{
+    if (GetModelType() != eModelInfoType::CLUMP)
+        return false;
+
+    CBaseModelInfoSAInterface*   pBaseObjectInfo = (CBaseModelInfoSAInterface*)ppModelInfo[usModelId];
+    CAtomicModelInfoSAInterface* m_pInterface = new CAtomicModelInfoSAInterface();
+    MemCpyFast(m_pInterface, pBaseObjectInfo, sizeof(CBaseModelInfoSAInterface));
```

Comment:
Copying of a whole class object is quite dangerous. I doubt that this class is a POD structure that can be copied without any issues.

Reply:
this pattern already exists in "void CModelInfoSA::MakeObjectModel(ushort usBaseID)" function